src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,151 @@
+package com.epam.lowcost.DAO;
+
+import com.epam.lowcost.model.Flight;
+import com.epam.lowcost.model.Plane;
+import com.epam.lowcost.util.DateFormatter;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlightDAOImpl implements FlightDAO {
+    private DataSource dataSource;
+
+    public FlightDAOImpl(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+
+    @Override
+    public List<Flight> getAllFlights() {
+        List<Flight> flights = new ArrayList<>();
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS JOIN  PLANES " +
+                     "ON FLIGHTS.plane_id = PLANES.id WHERE  FLIGHTS.ISDELETED = false ")) {
+            while (rs.next()) {
+                flights.add(extractFlightFromRS(rs));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flights;
+    }
+
+    @Override
+    public Flight getById(Long id) {
+        Flight flight = new Flight();
+        String sql = String.format("SELECT * FROM FLIGHTS JOIN  PLANES" +
+                " ON FLIGHTS.plane_id = PLANES.id  WHERE FLIGHTS.id = '%d' AND FLIGHTS.isDeleted=FALSE", id);
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery(sql)) {
+            if (rs.next()) {
+                return extractFlightFromRS(rs);
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flight;
+    }
+
+    @Override
+    public Flight addNewFlight(Flight flight) {
+        Long price = flight.getInitialPrice();
+        LocalDateTime depatureDate = flight.getDepartureDate();
+        LocalDateTime arrivalDate = flight.getArrivalDate();
+        Long plane_id = flight.getPlane().getId();
+
+        String sql = String.format("INSERT INTO Flights (initialPrice, plane_id, departureDate,arrivalDate,isDeleted)" +
+                        " VALUES (%d,%d,'%s','%s','%s')", price,
+                plane_id,
+                DateFormatter.format(depatureDate),
+                DateFormatter.format(arrivalDate),
+                "FALSE");
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS");
```

Comment:
you INSERT new flight, ok. What reason for SELECT * ? Maybe if INSERT finishes with success then you return the same object ?

Reply:
I need this to set id of flight object, which is autoincremented in database

Reply:
I have not noticed this in others before. Ok, enter in Google: `jdbc get id in case of insert`

Reply:
Please try resolve this without additional request to DB

Reply:
used generatedKeys method

Reply:
krasava